app/controllers/api/passwords_controller.rb

```diff
@@ -0,0 +1,29 @@
+class Api::PasswordsController < ApplicationController
+  skip_before_action :authenticate
+  respond_to :json
+
+  def create
+    if password_reset_token.present? && user.present?
+      user.update(password_params)
+      respond_with :api, user
```

Comment:
Here's a question: is ActiveRecord making the `User.find_by(password_reset_token)` call three times here, since you're not memoizing it?